concent_api/api-test.py

```diff
@@ -1,72 +1,152 @@
+            timestamp = current_time,
+            message_task_to_compute = message_task_to_compute_dumped
+        )
+
+    elif message_type == 'MessageAckReportComputedTask':
+        return MessageAckReportComputedTask(
+            timestamp = current_time,
+            message_task_to_compute = message_task_to_compute_dumped,
+        )
+    else:
+        assert False, 'Invalid message type'
+
+    return None
+
+
+def create_reject_data(task_id):
+    assert task_id > 0
+
     current_time = int(datetime.datetime.now().timestamp())
-    data = {
-        "type":         "MessageRejectReportComputedTask",
-        "timestamp":    current_time,
-        "reason":       "cannot-compute-task",
-        "message_cannot_commpute_task": {
-            "type":         "MessageCannotComputeTask",
-            "timestamp":    current_time,
-            "reason":       "provider-quit",
-            "task_id":      task_id
-        }
-    }
-    return data
+    message_cannot_compute_task = MessageCannotComputeTask(
+        timestamp = current_time,
+        reason = 'provider-quit',
+        task_id = task_id,
+    )
+
+    message_cannot_compute_task_dumped = dump(
+        message_cannot_compute_task,
+        CONCENT_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+
+    )
+
+    return MessageRejectReportComputedTask(
+        timestamp = current_time,
+        reason = 'cannot-compute-task',
+        message_cannot_compute_task = message_cannot_compute_task_dumped,
+    )
+
+
+def print_golem_message(message, indent = 4):
+    assert isinstance(message, Message)
+
+    HEADER_FIELDS  = ['timestamp', 'encrypted', 'sig']
+    PRIVATE_FIELDS = {'_payload', '_raw'}
+
+    assert 'type' not in message.__slots__
+    fields = ['type'] + HEADER_FIELDS + sorted(set(message.__slots__) - set(HEADER_FIELDS) - PRIVATE_FIELDS)
+    values = [
+        type(message).__name__ if field == 'type' else
+        getattr(message, field)
+        for field in fields
+    ]
+
+    for field, value in zip(fields, values):
+        if isinstance(value, bytes):
+            try:
+                nested_message = load(
+                    value,
+                    CONCENT_PRIVATE_KEY,
+                    CONCENT_PUBLIC_KEY,
+                )
+            except AttributeError:
+                # FIXME: golem-messages provides no reliable way to discern invalid messages from other AttributeErrors.
+                print('{}{:30} = <BINARY DATA>'.format(' ' * indent, field))
+            else:
+                print('{}{:30} ='.format(' ' * indent, field))
+                print_golem_message(nested_message, indent = indent + 4)
+        else:
+            print('{}{:30} = {}'.format(' ' * indent, field, value))
 
 
 def api_request(host, endpoint, data = None, headers = None):
-    if headers is None:
-        headers = DEFAULT_HEADERS
+    assert endpoint not in ['', None]
+    assert host     not in ['', None]
+    assert headers  not in ['', None]
```

Comment:
It can be changed to 
```
if not all([endpoint, host, headers]):
    raise Exception()
```
but also please remove `= None ` from function definition if it is not accepted.

Reply:
1) Yes, but `assert` is fine and let's be explicit about comparisons:

``` python
assert all(value not in ['', None] for value in [endpoint, host, headers]), "Some description."
```

2) (I think I talked about it with @Jakub89 before) I think that we should avoid having too many non-obvious defaults. It often causes hours lost in debugging down the line. And the most obvious default here is to simply have no headers.